File: archive/nodejs/muse/algorithms/search/BinarySearchTest.js
import * as SequenceBuilder from '../../util/SequenceBuilder.js';
import * as TestRunner from '../../util/TestRunner.js';
import * as BinarySearch from './BinarySearch.js';

const testBinarySearch = () => {
  const size = 32768;

  const arr = new Array(size);
  SequenceBuilder.packIncreasing(arr);

  if (BinarySearch.find(arr, -1) !== size) {
    return false;
  }

  if (BinarySearch.find(arr, 2_147_483_647) !== size) {
    return false;
  }

  for (let i = 0; i < size; i++) {
    if (BinarySearch.find(arr, arr[i]) !== i) {
      return false;
    }
  }

  return true;
};

const main = () => {
  TestRunner.pick(testBinarySearch);
};

main();

File: archive/ruby/muse/algorithms/search/BinarySearchTest.rb
require "muse/algorithms/search/BinarySearch"
require "muse/util/SequenceBuilder"
require "muse/util/TestRunner"

def testBinarySearch
  size = 32768

  arr = Array.new(size)
  SequenceBuilder.packIncreasing(arr)

  return false if BinarySearch.find(arr, -1) != size

  return false if BinarySearch.find(arr, 2_147_483_647) != size

  arr.each_index do |i|
    return false if BinarySearch.find(arr, arr[i]) != i
  end

  true
end

TestRunner.pick(testBinarySearch) if __FILE__ == $PROGRAM_NAME
